### PR TITLE
[openvino] update to 2023.2.0

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Test run

### CODE REVIEW

File: ports/openvino/usage [hunk 1]
```diff
@@ -2,3 +2,15 @@ The package OpenVINO can be used in cmake scripts as:
...
+for decent performance on discrete GPU devices. If you want to levarage performance of dGPU,
+please, consider either option:
+- Building with 'VCPKG_LIBRARY_LINKAGE=dynamic'
+- Disable Intel CPU plugin, because it uses different oneDNN flavor
+
+Note 2: If you are building OpenVINO as a set of shared libraries and want to use either Paddle, TensorFlow or ONNX frontends, it's important to build protobuf statically. For this, please, create new / modify existing triplet file with:
+
+if(PORT STREQUAL "protobuf")
+    set(VCPKG_LIBRARY_LINKAGE static)
+endif()
```
Comment (dg0yt) [inline ports/openvino/usage:16]: Why is this needed when all dependencies are using the same vcpkg protobuf package?

Comment (ilya-lavrenov) [reply ports/openvino/usage:16]: > all dependencies

which `all`?

dynamically OpenVINO does not work with dynamic protobuf. It's a OpenVINO + Protobuf limitation, because protobuf has some global variables..

Comment (dg0yt) [reply ports/openvino/usage:16]: So IIUC it is about modules/plugins (not about dynamic libraries in general), and about being unable to avoid registering the same extension (number) multiple times to a global registry while plugins are loaded at runtime. With the static protobuf in each plugin, you want to avoid the global registry.

Comment (ilya-lavrenov) [reply ports/openvino/usage:16]: Right

Comment (BillyONeal) [reply ports/openvino/usage:16]: Just for my own education: Does linking statically fix it because this global registry inside protobuf is a static variable or something, meaning that on all platforms if linked statically each `.so`/`.exe`/`.dll` gets their own copy of the global variable, and also on all platforms if linked dynamically they all share the same copy of that global variable?

Given what https://github.com/protocolbuffers/protobuf/issues/7826 says, should we have marked `protobuf` `ONLY_STATIC_LIBRARY`? (I'm not saying we would make that change now, I'm just trying to understand if this is always broken all the time or if the situation is broken specifically due to how OpenVINO loads plugins)

Comment (BillyONeal) [reply ports/openvino/usage:16]: Also https://github.com/protocolbuffers/protobuf/issues/7826#issuecomment-1210341403 suggests that newer protobuf may have resolved the issue?

Comment (ilya-lavrenov) [reply ports/openvino/usage]: Hi @BillyONeal 

> Does linking statically fix it because this global registry inside protobuf is a static variable or something, meaning that on all platforms if linked statically each .so/.exe/.dll gets their own copy of the global variable, and also on all platforms if linked dynamically they all share the same copy of that global variable?

Right.

> Given what https://github.com/protocolbuffers/protobuf/issues/7826 says, should we have marked protobuf ONLY_STATIC_LIBRARY?

It depends on the use case. For cases when libraries containing compiled proto files are not loaded dynamically - dynamic protobuf is OK. But in general, Protobuf team recommends to link protobuf statically to prevent multiple issues including an issue with different protobuf versions in a big single application (but since in vcpkg we can guarantee that protobuf is used of the same version across all packages, version-related issue does not seem to be a problem)

> Also https://github.com/protocolbuffers/protobuf/issues/7826#issuecomment-1210341403 suggests that newer protobuf (that vcpkg does have) may have resolved the issue?

The issue is still here, tried latest version.
Meanwhile, we have implemented WA for dynamic protobuf and ported from upstream in PR https://github.com/microsoft/vcpkg/pull/34651, where we hold libraries loaded dynamically for a while until the app is unloaded by OS, which allows to ensure compiled proto files are registered multiple times. It's still possible to imagine usecase which can lead to the original issue, but it's more rare usecase than a default one.
